### PR TITLE
[AutoParallel]:fix auto parallel grad merge bug

### PR DESCRIPTION
**Pull request by blacksheep-Aristotle**
### PR Category

Auto Parallel

### PR Types

Bug fixes

### Description

Pcard-67164

### CODE REVIEW

File: python/paddle/distributed/passes/auto_parallel_gradient_merge.py [hunk 1]
```diff
@@ -269,6 +269,44 @@ def _append_gradient_merge_backward_op(
...
+            # NOTE(zhangwl):temp only consider one operand op
+            move_to_opt_block_flag = False
+            return
+        for op_result in backward_op.results():
+            for next_op in op_result.all_used_ops():
+                if next_op.op_role != int(OpRole.Optimize):
+                    find_move_op(next_op)
+
+    find_move_op(used_grad_op)
+    if move_to_opt_block_flag:
```
Comment (From00) [inline python/paddle/distributed/passes/auto_parallel_gradient_merge.py:305]: move_to_opt_block_flag是否是一个冗余的判断？

File: python/paddle/distributed/passes/auto_parallel_gradient_merge.py [hunk 2]
```diff
@@ -269,6 +269,44 @@ def _append_gradient_merge_backward_op(
     return new_params_grads, grad_to_gradient_merge
 
 
+def _find_move2opt_ops(used_grad_op, grad):
```
Comment (From00) [inline python/paddle/distributed/passes/auto_parallel_gradient_merge.py]:
```suggestion
def _move_used_grad_op(used_grad_op, grad):
```

File: python/paddle/distributed/passes/auto_parallel_gradient_merge.py (hunk 1 above)
Comment (blacksheep-Aristotle) [reply python/paddle/distributed/passes/auto_parallel_gradient_merge.py:305]: 个人认为用这个flag比较直观

File: python/paddle/distributed/passes/auto_parallel_gradient_merge.py (hunk 2 above)
Comment (blacksheep-Aristotle) [reply python/paddle/distributed/passes/auto_parallel_gradient_merge.py]: done

**Review by From00 — APPROVED** [review]

**Pull request merged**